### PR TITLE
Add Base FamilyTree Contract

### PR DESCRIPTION
**Pull request by Gaonuk**
# Summary
- createHuman 
- addParentsToHuman
- getHuman
- createTree
- addMemberToFamilyTree
- mappings for humans
- mappings for trees
- (temporary) mapping for managers of humans and trees, to replace with NFT ownership later

Opened a PR to start discussing the code and improvements

### CODE REVIEW

**Comment by jjeangal** [comment]
We should change the names of father and mother to left and right parents.

**Comment by jjeangal** [comment]
Why is there both human managers and tree managers?

File: chain/FEVM-Hardhat-Kit/contracts/FamilyTree.sol [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        Human storage human = humans[id];
+        human.fatherId = leftParentId;
+        human.motherId = rightParentId;
+    }
+
+    function getHuman(uint256 humanId) external view returns (Human memory) {
+        return humans[humanId];
+    }
+
+    function createTree(uint256 id) external returns (uint256) {
```
Comment (jjeangal) [inline chain/FEVM-Hardhat-Kit/contracts/FamilyTree.sol:68]: where/how is the id computed?

Comment (Gaonuk) [reply chain/FEVM-Hardhat-Kit/contracts/FamilyTree.sol:68]: That id points towards the first member of the family tree

Comment (Gaonuk) [reply chain/FEVM-Hardhat-Kit/contracts/FamilyTree.sol:68]: For the treeId it is created at the bottom once we assign the first member
`uint256 treeId = uint256(keccak256(abi.encodePacked(familyMembersIds)));`

Comment (jjeangal) [reply chain/FEVM-Hardhat-Kit/contracts/FamilyTree.sol:68]: Great, sounds like a good workaround.

**Pull request merged**